### PR TITLE
Setup continuous integration and tdd environment 

### PR DESCRIPTION
**Pull request by tsirif**
Contains and sets up:
- **tox**
- setup.py commands for **pytest**
- **coverage** configuration.
- **test environments** (based on *tox*) for various Python versions that cover code by testing.
- **codecov** configuration
- tox commands and configurations for flake8, pylint, doc8, check-manifest, readme_renderer
- **sphinx docs and `conf.py`** + tox commands for building & serving to localhost
- helpers for building 📦 sdist and bdist_wheel, and distributing them to PyPI
- 👔 MANIFEST.in and .gitignore 🔥 
- sets up `.travis.yml` for Python matrix build

### CODE REVIEW

**Comment by nouiz** [comment]
At high level, it look good. Next time, make smaller PR, otherwise it is hard to review. Travis is failing.

**Comment by tsirif** [comment]
@nouiz Travis is set up, but it will continue to fail unless:
a. It has some tests to run. So this branch will make py27,py34,py35,py36,py37 builds to fail, because there are no tests currently.
b. Code is linted. However, linted code is in #7, not this branch.

The procedure works fine though. I need to merge this branch so that I can work on top of the environment.

File: .gitignore [hunk 1]
```diff
@@ -1,41 +1,76 @@
...
-*.out
-*.app
+# Translations
+*.mo
+*.pot
 
-build/*
-Build/*
+# Django stuff:
+*.log
```
Comment (bouthilx) [inline .gitignore]: Will we use Django? :confused:

File: setup.py [hunk 2]
```diff
@@ -1,66 +1,84 @@
...
+    long_description=textwrap.dedent(metaopt.__doc__),
+    license=metaopt.__license__,
+    author=metaopt.__author__,
+    author_email=metaopt.__author_email__,
+    url=metaopt.__url__,
+    packages=find_packages(where='src'),
+    package_dir={'': 'src'},
+    include_package_data=True,
+    data_files=find_data_files(),
+    scripts=['scripts/mopt'],
```
Comment (bouthilx) [inline setup.py]: I guess this will be changed by PR #7

Comment (tsirif) [reply setup.py]: idd

**Comment by nouiz** [comment]
You can add a dummy tests in this PR to make it pass. Also you can comment
the linter and rebase/uncomment it in pr 7.

Or do you prefer to merge 7 and rebase this one?

**Comment by tsirif** [comment]
I will put dummy tests, and rebase #7 on top!

**Comment by tsirif** [comment]
Might change to circle CI in the future, I cannot find a way to collect coverage from "all" matrix runs and upload that one to codecov.

Also, set up appveyor.

**Comment by nouiz** [comment]
For now, I think there is enough CI:)

**Pull request merged**